tests/e2e/helm-chart/chainsaw-test.yaml

```diff
@@ -0,0 +1,128 @@
+kind: Test
+metadata:
+  name: helm-chart
+spec:
+  description: This e2e test install Odigos via helm chart on custom namespace
+  skipDelete: true
+  steps:
+  - name: Prepare destination
+    try:
+      - script:
+          content: |
+            helm repo add grafana https://grafana.github.io/helm-charts
+            helm install e2e-tests grafana/tempo -n traces --create-namespace --set tempo.storage.trace.block.version=vParquet4
+      - assert:
+          file: assert-tempo-running.yaml
+  - name: Wait for destination to be ready
+    try:
+      - script:
+          timeout: 60s
+          content: ../../common/wait_for_dest.sh
+  - name: Install Odigos
+    try:
+    - script:
+        content: |
+          git clone https://github.com/odigos-io/odigos-charts.git /tmp/odigos-charts
+          # Retry 3 times to avoid flakiness
+          while ! helm upgrade --install odigos /tmp/odigos-charts/charts/odigos --create-namespace --namespace odigos-test-ns --set image.tag=e2e-test; do
+            echo "Failed to install Odigos, retrying..."
+            sleep 1
+          done
+          kubectl label namespace odigos-test-ns odigos.io/system-object="true"
+          rm -rf /tmp/odigos-charts
+        timeout: 60s
+  - name: Verify Odigos Installation
+    try:
+    - script:
+        content: |
+          export ACTUAL_VERSION=$(../../../cli/odigos version --cluster)
+          if [ "$ACTUAL_VERSION" != "e2e-test" ]; then
+            echo "Odigos version is not e2e-test, got $ACTUAL_VERSION"
+            exit 1
+          fi
+    - assert:
+        file: assert-odigos-installed.yaml
+  - name: Install Demo App
+    try:
+    - script:
+        timeout: 100s
+        content: |
+          docker pull keyval/odigos-demo-inventory:v0.1
+          docker pull keyval/odigos-demo-membership:v0.1
+          docker pull keyval/odigos-demo-coupon:v0.1
+          docker pull keyval/odigos-demo-inventory:v0.1
+          docker pull keyval/odigos-demo-frontend:v0.2
+          kind load docker-image keyval/odigos-demo-inventory:v0.1
+          kind load docker-image keyval/odigos-demo-membership:v0.1
+          kind load docker-image keyval/odigos-demo-coupon:v0.1
+          kind load docker-image keyval/odigos-demo-inventory:v0.1
+          kind load docker-image keyval/odigos-demo-frontend:v0.2
+    - apply:
+        file: 02-install-simple-demo.yaml
+    - assert:
+        file: assert-apps-installed.yaml
+  - name: Detect Languages
+    try:
+    - apply:
+        file: 03-instrument-ns.yaml
+    - assert:
+        file: assert-runtime-detected.yaml
+  - name: Add Destination
+    try:
+    - apply:
+        file: 04-add-destination.yaml
+    - assert:
+        file: assert-instrumented-and-pipeline.yaml
+  - name: Generate Traffic
+    try:
+      - script:
+          timeout: 60s
+          content: |
+            while true; do
+              # Apply the job
+              kubectl apply -f 05-generate-traffic.yaml
```

Comment:
Why do we flush traces?
And why todo this process multiple times? 
Don't we want to generate traffic and then just wait until the trace is ready?
Is it flaky or something?

Reply:
Good question: Tempo is sort of eventually consistent, meaning you don't immediately get traces you just inserted in a query.
I force flush it so they are shown in the query without having to sleep until tempo persists the data.